packages/@aws-cdk/aws-apigateway/lib/integrations/execution-input-builder.ts

```diff
@@ -0,0 +1,846 @@
+/**
+ * Builder to build execution input object
+ */
+export class ExecutionInputBuilder {
```

Comment:
What is this class? I don't get it. 

Looks very complicated. 

Why is this not a simple typescript interface?

Reply:
It is a string builder class for request context string building. It makes the code more readable.